test/gov/verify_surfaced_error.sh

```diff
@@ -0,0 +1,96 @@
+
+echo "Testing node error surfacing to exchange"
+
+IBM_ADMIN_AUTH="IBM/ibmadmin:ibmadminpw"
+KEY_TEST_DIR="/tmp/keytest"
+export HZN_EXCHANGE_URL="${EXCH_APP_HOST}"
+
+# cpu service - needed by the hzn dev tests and the location top level service as a 3rd level dependency.
+VERS="1.2.3"
+cat <<EOF >$KEY_TEST_DIR/svc_cpu.json
+{
+  "label":"CPU service",
+  "description":"CPU service",
+  "public":true,
+  "url":"https://bluehorizon.network/service-cpu",
+  "version":"$VERS",
+  "arch":"amd64",
+  "sharable":"singleton",
+  "matchHardware":{},
+  "userInput":[
+    {
+      "name":"cpu_var1",
+      "label":"",
+      "type":"string"
+    }
+  ],
+  "deployment":{
+    "services":{
+      "cpu":{
+        "image":"openhorizon/example_ms_x86_cpu:1.2.2",
+        "binds":["/tmp:/hosttmp",""]
+      }
+    }
+  },
+  "deploymentSignature":""
+}
+EOF
+echo -e "Re-register IBM/cpu service with deployment error $VERS:"
+hzn exchange service publish -I -O -u $IBM_ADMIN_AUTH -o IBM -f $KEY_TEST_DIR/svc_cpu.json -k $KEY_TEST_DIR/*private.key
+if [ $? -ne 0 ]
+then
+    echo -e "hzn exchange service publish failed for IBM/cpu."
+    exit 2
+fi
+
+
+
+NUM_ERRS=0
+TIMEOUT=0
+while [ $NUM_ERRS -le 0 ] && [ $TIMEOUT < 15 ]
+do
+  ERRS=$(curl -sSL http://localhost/eventlog/surface)
+  NUM_ERRS=$(echo ${ERRS} | jq -r '. | length')
+  sleep 5s
+  TIMEOUT = $TIMEOUT+1
+done
+
+echo -e "Found surfaced error $ERRS"
+
+# cpu service - needed by the hzn dev tests and the location top level service as a 3rd level dependency.
+VERS="1.2.2"
```

Comment:
I suspect this needs to be 1.2.4 in order to get the error to stop. And I think you need to add some checking that the errors have stopped.